rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -132,12 +133,17 @@ rcl_logging_ret_t rcl_logging_external_initialize(
       return dir_ret;
     }
 
-    // SPDLOG doesn't automatically create the log directories, so create them
-    rcpputils::fs::path logdir_path(logdir);
-    if (!rcpputils::fs::create_directories(logdir_path)) {
-      RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING("Failed to create log directory: %s", logdir);
-      allocator.deallocate(logdir, allocator.state);
-      return RCL_LOGGING_RET_ERROR;
+    std::error_code ec;
+    std::filesystem::path logdir_path(logdir);
+    // std::filesystem::create_directory returns fail when the target directory is already exists.
+    // So, check it first.
+    if (!std::filesystem::is_directory(logdir_path, ec)) {
+      // SPDLOG doesn't automatically create the log directories, so create them
+      if (!std::filesystem::create_directories(logdir_path, ec)) {
```

Comment:
https://en.cppreference.com/w/cpp/filesystem/create_directory 5 & 6 says create_director**ies** creates the directories if they don't exist otherwise it does nothing, so maybe we don't need to check if there's an existing directory at all.